Missing license in gemspec (and RubyGems)

The published `openai` gem is missing the license field in its gemspec file, which prevents the license information from being displayed on the gem's RubyGems page and causes failures in automated compliance audits.

## Current Behavior
- The gem's RubyGems page at https://rubygems.org/gems/openai shows no license information
- The repository clearly states the code is under Apache-2.0 license
- The LICENSE file exists in the repository

https://rubygems.org/gems/openai
<img width="280" height="231" alt="Image" src="https://github.com/user-attachments/assets/a88be3c6-eb76-43a4-9d23-d9b8edc10d52" />

`license_finder` result:
```
Dependencies that need approval:
openai, 0.31.0, unknown
Error: Process completed with exit code 1.
```

## Expected Behavior
The gemspec should include the license field so that:
1. The license is properly displayed on RubyGems.org
2. Automated tools can properly detect the gem's license
3. Users can verify licensing compliance programmatically

## Suggested Fix
Add the following line to `openai.gemspec`:
```ruby
spec.license = "Apache-2.0"
```

## Why This Matters
- **Compliance**: Many organizations require license information to be explicitly declared for dependency scanning (`license_finder` gem)
- **Transparency**: Users should be able to see the license information directly on RubyGems.org
- **Best Practices**: Including the license in the gemspec is a Ruby community best practice
- **Tooling**: Dependency management tools (like Bundler's `bundle licenses` command) rely on this field

--- a/openai.gemspec
+++ b/openai.gemspec
@@ -13,6 +13,7 @@ Gem::Specification.new do |s|
   s.metadata["source_code_uri"] = "https://github.com/openai/openai-ruby"
   s.metadata["rubygems_mfa_required"] = false.to_s
   s.required_ruby_version = ">= 3.2.0"
+  s.license = "Apache-2.0"
 
   s.files = Dir[
     "lib/**/*.rb",

